### PR TITLE
Log4j2 move

### PR DESCRIPTION
**Pull request by fxprunayre**
Hi @jodygarnett this will fix the log file download from the harvester history panel.
I moved to the GeonetworkDataDirectory the method to setup the log file so we can access from the modules which need it. For the time being I also commented out the checked we added some times ago when harvester files where not existing. This in HarvesterHistory domain module and I'm not sure what would be the proper way to fix that. Any suggestions welcomed. Thanks

image

### CODE REVIEW

File: harvesters/src/main/java/org/fao/geonet/services/harvesting/Log.java [hunk 1]
```diff
@@ -62,20 +64,23 @@ public Element exec(Element params, ServiceContext context)
...
                     + logfile);
         }
 
-        File file = new File(logfile);
 
-        if (!file.exists() || !file.canRead()) {
-            throw new NullPointerException(
-                "Couldn't find or read the logfile. Somebody moved it? " + file.getAbsolutePath());
+        File mainLogFile = GeonetworkDataDirectory.getLogfile();
+        Path file = mainLogFile.toPath().getParent().resolve(logfile);
```
Comment (josegar74) [inline harvesters/src/main/java/org/fao/geonet/services/harvesting/Log.java:75]: ` GeonetworkDataDirectory.getLogfile()` can return `null`.

File: domain/src/main/java/org/fao/geonet/domain/HarvestHistory.java [hunk 2]
```diff
@@ -311,29 +297,30 @@ public Element getInfoAsXml() throws IOException, JDOMException {
...
+//                        iter.remove();
+//                    } else {
+//                        isLogFileFound = true;
+//                    }
+//                } else {
+//                    Log.debug(Constants.DOMAIN_LOG_MODULE, "Harvest history logfile `" + path + "` ignored, no longer available");
+//                    iter.remove();
+//                }
+//            }
+//        }
```
Comment (josegar74) [inline domain/src/main/java/org/fao/geonet/domain/HarvestHistory.java:323]: Can this cause any issue or it's just to store the log file content in the database?

Comment (jodygarnett) [reply domain/src/main/java/org/fao/geonet/domain/HarvestHistory.java:323]: A lot of this code was to recover from the times when multiple logfiles were listed, not sure why that is being commented out.

File: harvesters/src/main/java/org/fao/geonet/services/harvesting/Log.java (hunk 1 above)
Comment (jodygarnett) [reply harvesters/src/main/java/org/fao/geonet/services/harvesting/Log.java:75]: will use new Path(logfile) if needed then

**Pull request merged**